internal/fs/fs_test.go

```diff
@@ -115,6 +116,9 @@ func (t *fsTest) SetUpTestSuite() {
 	mtimeClock = timeutil.RealClock()
 	cacheClock.SetTime(time.Date(2015, 4, 5, 2, 15, 0, 0, time.Local))
 	t.serverCfg.CacheClock = &cacheClock
+	if bucketType == gcs.Nil {
+		bucketType = gcs.NonHierarchical
```

Comment:
A new global variable, bucketType, has been added to modify the type of bucket used in tests. Its default value is nil, and logic has been implemented to treat it as NonHierarchical if it's not explicitly defined. For rest of the tests we trigger NonHierarchical bucket(existing) flow.

For HNS tests, the bucketType is set to Hierarchical so it will trigger Hierarchical bucket flow.

Reply:
Got it. Hns_test is calling setup method of fs_test.